sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -84,10 +79,31 @@ package object debug {
     }
     codegenSubtrees.toSeq.map { subtree =>
       val (_, source) = subtree.doCodeGen()
-      (subtree.toString, CodeFormatter.format(source))
+      (subtree, source)
     }
   }
 
+  /**
+   * Get WholeStageCodegenExec subtrees and the codegen in a query plan
+   *
+   * @param plan the query plan for codegen
+   * @return Sequence of WholeStageCodegen subtrees and corresponding codegen
+   */
+  def codegenStringSeq(plan: SparkPlan): Seq[(String, String)] = {
+    codegenSubtreeSourceSeq(plan).map(s => (s._1.toString, CodeFormatter.format(s._2)))
+  }
+
+
+  /**
+   * Get WholeStageCodegenExec subtrees' CodeAndComment in a query plan
+   *
+   * @param plan the query plan for CodeAndComment
+   * @return Sequence of WholeStageCodegen subtrees' `CodeAndComment`
+   */
+  def codegenCodeAndCommentSeq(plan: SparkPlan): Seq[CodeAndComment] = {
+    codegenSubtreeSourceSeq(plan).map(_._2)
```

Comment:
we essentially need the changes in this file? I feel it's okay to touch `TPCDSQuerySuite.scala` only like: https://github.com/apache/spark/compare/master...maropu:SPARK-22140-FOLLOWUP

Reply:
Thanks, we can do. I wanted to reduce duplicated code.

Reply:
probably, the debug package is not for test uses like this.